pkg/util/machines.go

```diff
@@ -114,12 +114,24 @@ func IsControlPlaneMachine(machine metav1.Object) bool {
 // GetMachineMetadata returns the cloud-init metadata as a base-64 encoded
 // string for a given VSphereMachine.
 func GetMachineMetadata(hostname string, vsphereVM infrav1.VSphereVM, networkStatuses ...infrav1.NetworkStatus) ([]byte, error) {
+	return GetMachineMetadataWithIPAMState(hostname, vsphereVM, map[string]infrav1.NetworkDeviceSpec{}, networkStatuses...)
+}
+
+// GetMachineMetadata returns the cloud-init metadata as a base-64 encoded
+// string for a given VSphereMachine.
+// IPAM state includes IP and Gateways that should be added to each device.
```

Comment:
Update the comment and consider making it a private method if not used directly outside the package.

Reply:
We get apidiff failures when we reduce the visibility of functions in this package. Is that ok? I don't think the functions in this package are used widely. I'm not a big fan of the factoring. The functions are public so that other packages can use them, but they're really internal functions.

Reply:
I meant update the first word in the comment to correctly reflect the name of the method
~~GetMachineMetadata~~ --> GetMachineMetadataWithIPAMState

Reply:
Ahhh, didn't see it. Fixed!